src/data.rs

```diff
@@ -39,6 +80,13 @@ impl Warp10Serializable for Value {
     }
 }
 
+impl Value {
+    #[cfg(feature = "json")]
+    pub fn try_from(obj: &impl Serialize) -> error::Result<Self> {
```

Comment:
Why not just use the TryFrom trait?

Reply:
If for some reason we go with this instead of TryFrom, I'd prefer if we had `&dyn` instead of `&impl`

Reply:
I tried multiple implementations, starting by adding an entry to the `enum Value` which was `Json(Box<dyn Serialize>)`, but I got the following error `Serialize cannot be made into an object`. 

So, I tried to fix the error using a library named `erased-serde` which take care of this issue by shadowing the object behind a pointer. So, that the rust compiler let me compile, but I got a second issue with the `PartialEq` trait in the derive macro which required the object to be `Sized` as well. 

To that point, I wonder that I do not implement this issue in the right way, because the `Serialize` trait required the `Sized` trait on the object at the compilation. From here, I started using the `impl` keyword which act as a generic resolved at the compilation and from here, it was easier to implement this feature. 

I agree with you that using the `TryFrom` trait will be the right way to do it, but it not allowed to write something like this:

```rust
impl TryFrom<&impl Serialize> for Value {
```

May be I miss a something and if you see a better way to achieve it using the `dyn` keyword, I will be glad to use it :).